### DISCUSSION TITLE
How to set the a expiry date in firebase dynamic link ?

### DISCUSSION THREAD

**Discussion by MabudAlam** [firebase/flutterfire#10658] [Q&A]
navigationInfoParameters: NavigationInfoParameters(
      forcedRedirectEnabled: true,
      enableForcedRedirect: true,
     expiryDate: expiryDate,      //expiry date is missing in the plugin
    ),

**Comment by Vaytrex** [comment]
For anyone landing here after the Aug 2025 shutdown: there's no drop-in. You need a URL that branches on User-Agent (App Store / intent:// with browser_fallback_url / website) and skips redirecting crawlers so previews survive. Full write-up with the intent format: https://linkbah.app/firebase-dynamic-links-alternative

**Comment by mohamadnoorir63-svg** [comment]
Title
Google Sign-In fails only in Google Play distributed app (ApiException: 10), but Release APK works perfectly
Body
Hello,
I have been trying to solve this issue for several days without success.
Environment
Flutter 3.32.2
firebase_core 3.15.2
firebase_auth 5.7.0
google_sign_in 6.3.0
google_sign_in_android 6.2.1
Package name:
com.heratnoor.afghanmarket
Firebase project:
afghanbazar-a5ac4
The problem
Google Sign-In works perfectly when I install a locally built Release APK.
I build it using:
flutter clean
flutter pub get
flutter build apk --release
After installing the APK manually, Google Sign-In works correctly.
However, when I build an App Bundle:
flutter clean
flutter pub get
flutter build appbundle --release
upload it to Google Play, publish it, then install the application from Google Play, Google Sign-In always fails with:
PlatformException(sign_in_failed,
com.google.android.gms.common.api.ApiException: 10)
The Google account picker appears normally.
After selecting an account, sign-in immediately fails.
What has already been verified
Firebase Authentication (Google provider) is enabled.
Package name is correct.
OAuth clients exist.
google-services.json is correct.
AndroidManifest.xml checked.
Gradle configuration checked.
Release signing configuration checked.
AAB SHA-1 verified.
Upload keystore verified.
SHA fingerprints verified.
APK and AAB are built from exactly the same source code.
Testing
The issue has been tested on three different Android devices.
Results:
✅ Release APK works on all devices.
❌ Google Play installed version fails on all devices.
At this point I cannot identify any configuration difference between the APK and the Play-distributed version.
Has anyone experienced a case where Google Sign-In only fails in the Google Play version with ApiException: 10, while the Release APK works correctly?
Any guidance would be greatly appreciated.
Thank you.
